OpenRA.Mods.Common/Lint/CheckSequences.cs

```diff
@@ -19,15 +19,48 @@ class CheckSequences : ILintPass
 	{
 		public void Run(Action<string> emitError, Action<string> emitWarning, Map map)
 		{
-			var sequences = MiniYaml.MergeLiberal(map.SequenceDefinitions, Game.ModData.Manifest.Sequences.Select(s => MiniYaml.FromFile(s)).Aggregate(MiniYaml.MergeLiberal));
+			var sequenceDefinitions = MiniYaml.MergeLiberal(map.SequenceDefinitions,
+				Game.ModData.Manifest.Sequences.Select(s => MiniYaml.FromFile(s)).Aggregate(MiniYaml.MergeLiberal));
 
 			foreach (var actorInfo in map.Rules.Actors)
+			{
 				foreach (var renderInfo in actorInfo.Value.Traits.WithInterface<RenderSimpleInfo>())
 				{
 					var image = renderInfo.Image ?? actorInfo.Value.Name;
-					if (!sequences.Any(s => s.Key == image.ToLowerInvariant()) && !actorInfo.Value.Name.Contains("^"))
+					if (!sequenceDefinitions.Any(s => s.Key == image.ToLowerInvariant()) && !actorInfo.Value.Name.Contains("^"))
 						emitWarning("Sprite image {0} from actor {1} has no sequence definition.".F(image, actorInfo.Value.Name));
 				}
+
+				foreach (var traitInfo in actorInfo.Value.Traits.WithInterface<ITraitInfo>())
+				{
+					foreach (var field in traitInfo.GetType().GetFields())
+					{
+						if (field.HasAttribute<SequenceReferenceAttribute>())
+						{
+							var sequences = LintExts.GetFieldValues(traitInfo, field, emitError);
+							foreach (var sequence in sequences)
+							{
+								if (string.IsNullOrEmpty(sequence))
+									continue;
+
+								var renderInfo = actorInfo.Value.Traits.WithInterface<RenderSpritesInfo>().FirstOrDefault();
+								if (renderInfo == null)
+									continue;
+
+								var image = renderInfo.GetImage(actorInfo.Value, map.SequenceProvider, null); // TODO: check all possible races
```

Comment:
I now look up an image reference override so this won't need a hard-coded one anymore.